drivers/overlay/encryption.go

```diff
@@ -224,7 +225,9 @@ func programSA(localIP, remoteIP net.IP, spi *spi, k *key, dir int, add bool) (f
 	if add {
 		action = "Adding"
 		xfrmProgram = ns.NlHandle().XfrmStateAdd
-		crypt = &netlink.XfrmStateAlgo{Name: "cbc(aes)", Key: k.value}
+		salt := make([]byte, 4)
+		binary.LittleEndian.PutUint32(salt, k.tag)
+		crypt = &netlink.XfrmStateAlgo{Name: "rfc4106(gcm(aes))", Key: append(k.value, salt...), ICVLen: 64}
```

Comment:
Going back to this, I looked again the rfc4106 and the crypto code in kernel, and my understanding is in line with your previous thought:

> I assume we're not responsible for choosing the 8 octet IV?

which is the IV generation and per-key uniqueness is a concern of the cipher's implementer, not something we can control using the xfrm APIs.